contracts/clients/src/signer/getPublicKeyStr.ts

```diff
@@ -0,0 +1,16 @@
+import * as hubbleBls from "../../deps/hubble-bls";
+import getPublicKey from "./getPublicKey";
+
+export default (
+    signerFactory: hubbleBls.signer.BlsSignerFactory,
+    domain: Uint8Array,
+  ) =>
+  (privateKey: string): string => {
+    const [x0, x1, y0, y1] = getPublicKey(signerFactory, domain)(privateKey);
+    return hubbleBls.mcl.dumpG2([
+      x0.toString(),
+      x1.toString(),
+      y0.toString(),
+      y1.toString(),
```

Comment:
The `.toString()`s are a bit odd here... not sure if you realized. I want to explain what's going on here.

The result of `getPublicKey` is understandably typed as `PublicKey`, which is the wide/permissive type we get from typechain, so the elements are `BigNumberish`.

`dumpG2` requires strings, and so `.toString()`, being available on all possibilities of `BigNumberish` satisfies the type checker. However, `dumpG2` actually expects _hex_ strings (right?), so this will only work if all the `BigNumberish` elements are actually already hex strings.

...which they are, because they come from `signer.pubkey`.

tl;dr - `.toString()` satisfies the type checker, but the potential issue flagged by the type checker isn't actually fixed by `.toString()`.

I'm nit picking. I think. It's unlikely this will ever be an issue, but it's a cause of confusion and could be the cause of a bug with subsequent changes that would look reasonable.

I think I'll submit a followup PR that replaces the use of `getPublicKey` with the underlying hubblebls call. Using `as string` would also be preferable, since at least it keeps the flag that something odd is going on (because it is), and avoids the no-op.

Reply:
https://github.com/jzaki/bls-wallet/issues/83

Reply:
(Btw it's very possible that I also used the same kind of `.toString()` in the past 😅.)

Reply:
> However, dumpG2 actually expects hex strings (right?), so this will only work if all the BigNumberish elements are actually already hex strings. ...which they are, because they come from signer.pubkey.

(dumpG2)[https://github.com/jzaki/bls-wallet/blob/e366d0af4c339db3f380732550c9319ea4d6982b/contracts/clients/deps/hubble-bls/mcl.ts#L223-L226] explicitly needs the pubkey segments to be strings, and I imagine in hex format. Good eye, there are some cases where this will fail spectacularly if it is a `number` or `BigNumber`:
![Screenshot from 2022-01-04 14-01-37](https://user-images.githubusercontent.com/10394161/148123770-a255f06e-26ce-4d8f-90b3-aeca8c80eaf5.png)

We might want to just have the type for our pubkey be `[string, string,string, string]`, and then we can have a function that converts from the typechain type to our needed type (I don't think `hubble-bls` exports `solG2`). We could also just make a seperate wrapper class for it and put needed utility functions in that class similar to https://github.com/thehubbleproject/hubble-contracts/blob/master/ts/pubkey.ts

Reply:
Perhaps relevant here, some conversions remove preceding 0's from hex, which is problematic when a smart contract call requires the right length. Eg a 20 byte address:
`0x0102030405060708091011121314151617181920`, becoming,
`0x102030405060708091011121314151617181920`